nav2_msgs/action/FollowWaypoints.action

```diff
@@ -1,4 +1,6 @@
 #goal definition
+uint32 number_of_loops
+uint32 goal_index
```

Comment:
What's `goal_index`? This doesn't seem to me like information to have sent

Reply:
It denotes the waypoint that the robot is currently in. Especially when you pause the waypoint navigation from the nav2_panel, you need to somehow let the waypoint follower know about the last waypoint that the robot was navigating to. Because, when you pause the navigation, it typically means you cancel the navigation. So, we store the information about the goal index through the feedback and we send it again to the waypoint follower. If you see this [line](https://github.com/ros-planning/navigation2/blob/634d2e3d9b6bde5558c90fe6583d52b3ed66cf55/nav2_waypoint_follower/src/waypoint_follower.cpp#L173) from the original code, the goal index is always set to 0. Now, we just make it a bit flexible by adding an option for the FollowWaypoints action, so that we get an option to define which waypoint the robot should navigate to and of course gives us the the ability to pause and resume the waypoint navigation.

This may also help a scenario, where an user has 100 waypoints. He just does not want his robot to start from the first waypoint, but from 30th waypoint and then navigate to the rest of the 70 waypoints.

Reply:
That should not be in the request field - the first waypoint in the request is what it should do. I think this is a reason to not have the pause option

Reply:
Alright, let me see if I can find a workaround.